File: ROS_SERVICE/msg/Service_Client.py

import rospy
from demo_package import Pesan

def vector_magnitude_client(x, y, z):
    rospy.wait_for_service('calculate_magnitude')
    try:
        calculate_magnitude = rospy.ServiceProxy('calculate_magnitude', VectorService)
        response = calculate_magnitude(x, y, z)
        rospy.loginfo(f"Magnitude : {response}")
    except rospy.ServiceException as e:
        rospy.logerr(f"GAGAL: {e}")

if __name__ == "__main__":
    rospy.init_node('vector_magnitude_client')
    vector_magnitude_client(3.0, 4.0, 5.0)  # Contoh Nilai 

File: ROS_SERVICE/msg/Service_Client.cpp
#include "ros/ros.h"
#include "demo_package/Pesan.h"

int main(int argc, char **argv)
{
    ros::init(argc, argv, "vector_magnitude_client_cpp");
    ros::NodeHandle n;

    ros::ServiceClient client = n.serviceClient<my_ros_package::VectorService>("calculate_magnitude");
    my_ros_package::VectorService srv;
    
    srv.request.komponenX = 3.0;  // contoh nilai 
    srv.request.komponenY = 4.0;
    srv.request.komponenZ = 5.0;

    if (client.call(srv))
    {
        ROS_INFO("Magnitude: %f", srv.response.magnitude);
    }
    else
    {
        ROS_ERROR("Gagal");
        return 1;
    }

    return 0;
}
